pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1725,6 +1723,19 @@ public RealtimeSegmentDataManager(SegmentZKMetadata segmentZKMetadata, TableConf
     }
   }
 
+  // Consumption while downloading and replacing the slow replicas is not allowed for the following use cases:
+  // 1. Partial Upserts
+  // 2. Dedup Tables
+  // For the above table types, we would be looking into the metadata information when inserting a new record,
+  // so it is not ideal to allow consumption when downloading and replacing the consuming segment as we might see
+  // unusual behaviour of duplicates in dedup tables and inconsistent entries compared to lead replicas for partial
+  // upsert tables.
```

Comment:
nit: Consumption while downloading and replacing the `consuming segment` on slow replicas ...

nit: ... so it's not right to allow consumption .. as we might see duplicates in dedup tables or inconsistent entries in partial upsert table